### ISSUE TITLE
[Proposal] Monitor gin with https://github.com/alibaba/opentelemetry-go-auto-instrumentation

### ISSUE THREAD

**Issue by 123liuziming** [gin-gonic/gin#4107]
Hi all, As we all know gin is the best http framework in the golang ecosystem. But I found that gin doesn't have the ability to integrate tracing and metrics by default and there doesn't seem to be a best practice in the documentation on how to monitor gin applications. Our project(https://github.com/alibaba/opentelemetry-go-auto-instrumentation) allows tracing and metrics to be collected and reported on the gin app without making changes to user code. I was wondering if it would be possible for me to add an observable subsection to the official documentation for this project that describes best practices for observing the gin application. Thanks!

**Comment by FarmerChillax** [comment]
I hope the observable part of the document will focus on the otel project, supplemented by the alibaba project

**Comment by 123liuziming** [comment]
> supplemented

I agree, the project will be donated into OpenTelemetry Community soon, see https://github.com/open-telemetry/community/issues/1961. We will focus on the otel project.

**Comment by damemi** [comment]
We could also mention the OpenTelemetry eBPF based auto-instrumentation (https://github.com/open-telemetry/opentelemetry-go-instrumentation/), which supports Gin as well.

**Comment by 123liuziming** [comment]
> We could also mention the OpenTelemetry eBPF based auto-instrumentation (https://github.com/open-telemetry/opentelemetry-go-instrumentation/), which supports Gin as well.

Yes, I will raise a PR to introduce different ways of monitoring Gin Application. @appleboy What do you think about that?

**Comment by 123liuziming** [comment]
And can we also add a middleware for OpenTelemetry in `https://github.com/gin-contrib`?

**Comment by appleboy** [comment]
Also reference: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

**Comment by appleboy** [comment]
@123liuziming Thanks for your contribution. I closed the issue when PR was merged.
